### PR TITLE
MC-1268 Adding operations for configuration reload & update

### PR DESCRIPTION
**Pull request by erosb**
Fixes [MC-1268](https://hazelcast.atlassian.net/browse/MC-1268)

Adds `reloadConfig` and `memberConfig` operations. These are remote interfaces for the `ConfigurationService#update()` methods implemented as part of [HZ-760](https://hazelcast.atlassian.net/browse/HZ-760).

OS PR: https://github.com/hazelcast/hazelcast/pull/20529
EE PR: https://github.com/hazelcast/hazelcast-enterprise/pull/4621

### CODE REVIEW

File: protocol-definitions/MC.yaml
Comment (ramizdundar) [inline protocol-definitions/MC.yaml:879]: minor: Reloads -> Updates

File: protocol-definitions/custom/Custom.yaml
Comment (ramizdundar) [inline protocol-definitions/custom/Custom.yaml:1022]: Name can be null.

File: protocol-definitions/MC.yaml
Comment (promanenko) [inline protocol-definitions/MC.yaml:869]: do we need to return a result of the update here?
We are going to get status events anyway, may be it makes sense to have this process async

File: protocol-definitions/MC.yaml
Comment (promanenko) [inline protocol-definitions/MC.yaml:871]: we need to get information about the error in case of configuration update failure

Comment (erosb) [reply protocol-definitions/MC.yaml:871]: If we make this async (as per your other comment), then the failure reason will be included in the `ConfigUpdateFailedEvent` we receive on the MC side, won't it?

Comment (promanenko) [reply protocol-definitions/MC.yaml:871]: right

Comment (erosb) [reply protocol-definitions/MC.yaml:871]: ok thanks I will rework the PRs to make this operation async

**Review by mdumandag — APPROVED** [review]
Looks OK from the protocol's point of view.

We can merge this once the Hazelcast PRs are merged

**Review by promanenko — APPROVED** [review]

**Pull request merged**
